yt/data_objects/static_output.py

```diff
@@ -654,7 +655,18 @@ def print_stats(self):
     def field_list(self):
         return self.index.field_list
 
+    @property
+    def field_info(self):
+        if self._field_info is None:
+            self.create_field_info()
+        return self._field_info
+
+    @field_info.setter
+    def field_info(self, value):
+        self._field_info = value
+
     def create_field_info(self):
+        self.index
```

Comment:
I'm going to admit that this isn't a great thing to do as instantiating the index called `create_field_info`. What I think would be better is renaming this `_create_field_info` to remove the expectation that anyone ever call it. I would then change line 661 above to simply `self.index`.

Reply:
I've refactored this so that we only ever do one pass through `create_field_info`.